packages/@aws-cdk/assertions/lib/private/section.ts

```diff
@@ -0,0 +1,56 @@
+import { Matcher, MatchResult } from '../matcher';
+
+export type MatchSuccess = { match: true, matches: any[] };
+export type MatchFailure = { match: false, closestResult?: MatchResult, analyzedCount: number };
+
+export function matchSection(section: any, matcher: Matcher): MatchSuccess | MatchFailure {
+  let closestResult: MatchResult | undefined = undefined;
+  let matching: any[] = [];
+  let count = 0;
+
+  eachEntryInSection(
+    section,
+
+    (entry) => {
+      const result = matcher.test(entry);
+      if (!result.hasFailed()) {
+        matching.push(entry);
+      } else {
+        count++;
+        if (closestResult === undefined || closestResult.failCount > result.failCount) {
```

Comment:
`~~` Super nit. This reads better to me for some reason. Feel free to ignore.
```suggestion
        if (closestResult === undefined || result.failCount < closestResult.failCount) {
```